### ISSUE TITLE
[UI] Prompt for logging in immediately when using `taskcluster signin`.

### ISSUE THREAD

**Issue by tomprince** [taskcluster/taskcluster#2679]
**Describe the bug**
When signing in via `taskcluster sign`, if you are not signed via the web (or the signin has expired), it gives you a page that says "sign in to create client". You need to then click in the top right to sign in. 

**Expected behavior**

It would be nice if instead, it automatically prompted you to sign in, rather that requiring the extra click.

**Comment by helfi92** [comment]
Agreed, that would be good to have.

**Comment by ashishzingh** [comment]
i would like to work on this

**Comment by helfi92** [comment]
Go for it.

**Comment by cerkiewny** [comment]
Can I pick this one up? there was no activity on it for a long time,

I have been looking into what this would involve, I can either create a state handler and another dialogue in the component, use something like redux to propagate the state of visibility of login dialogue "up the chain" or use context to show it, what is prefered way? @helfi92

**Comment by jwhitlock** [comment]
Yes @cerkiewny, I've reassigned to you. I don't have any advice on how to implement this feature. I think @tomprince is asking for the "Sign In" dialog to automatically open, rather than requiring a click, but other ways to display the login options would also work.

I followed the instructions on the [Taskcluster Client for Shell](https://github.com/taskcluster/taskcluster/tree/main/clients/client-shell) README and downloaded the `taskcluster` tool. I had to set the `TASKCLUSER_ROOT_URL` environment variable, so it would be something like this for community:

```bash
set TASKCLUSTER_ROOT_URL=https://community-tc.services.mozilla.com/
taskcluster signin
```

This printed:

```
INFO Starting                                     
INFO Listening for a callback on: http://localhost:59193 
INFO Opening URL: https://community-tc.services.mozilla.com/auth/clients/create?scope=%2A&name=cli-ABC123&expires=1d&callback_url=http%3A%2F%2Flocalhost%3A59193&description=Temporary+client+for+use+on+the+command+line 
```

and opened a page showing the text "Sign in to create a client":

![tc_issue_2679](https://user-images.githubusercontent.com/286017/126560281-08edfcb3-c768-4215-ac3c-0898649f9eb4.png)

I then clicked "Sign In" and the modal dialog box popped up:

![tc_issue_2679_clicked](https://user-images.githubusercontent.com/286017/126561107-72d868a9-1b66-4f90-8a3e-abf61f65315f.png)

I logged in with GitHub, and I returned to the page with a form to create a new client:

![tc_issue_2679_create_client_mod](https://user-images.githubusercontent.com/286017/126561859-dcdaad7e-7dec-4136-94f8-5cff88daddae.png)

When I saved it, I was redirected to my localhost port.

![tc_issue_2679_final_mod](https://user-images.githubusercontent.com/286017/126561878-33f0cbbf-43dc-4a45-8d2f-83b4d7b87e36.png)

The command-line program then outputted a set of bash variables:

```bash
export TASKCLUSTER_CLIENT_ID='github/123456789|jwhitlock/cli-ABC123'
export TASKCLUSTER_ACCESS_TOKEN='RmFrZVNlY3JldEZvckdpdGh1Yklzc3Vl'
export TASKCLUSTER_ROOT_URL='https://community-tc.services.mozilla.com/'
INFO Credentials output as environment variables 
```

If I did it while signed in, it skipped to the part where it creates the new client credentials.

**Comment by cerkiewny** [comment]
I am on holidays this week and will pick this up next week.
I will probably make the dialogue be sort context provider so I can import the context and use its showLoginDialog function without passing through multiple layers of componetns.
